### ISSUE TITLE
Software Design: Use-Case Diagrams - Chat

### ISSUE THREAD

**Issue by egemenatikk** [bounswe/bounswe2022group2#76]
### Issue Description

* This is a sub issue of the #70.

* The use case diagram for chat and actions related to it will be created under this issue.

* The [requirements](https://github.com/bounswe/bounswe2022group2/wiki/Requirements) to be covered by these use cases will be shared here.

* The use case diagram will be generated using the Lucichart app, and the related link will be shared here and under #70.

### Final Actions

* The Use Case Diagram to be generated here must be combined with the use case diagrams of the other team members at #70. 

### Deadline of the Issue

02.04.2022 - Saturday - 14.00

### Deadline for the Review

02.04.2022 - Saturday - 15.00

**Comment by egemenatikk** [comment]
I have scanned our requirements page and decided that the following sub-entries are related to chat:

* 1.1.3.1.4. Users shall be able to share notes with each other.
* 1.1.3.2.5.1. Users who are qualified as lecturers shall create community events.
* 1.2.2.2 The system shall notify users regarding incoming messages from their communication channels.
* 1.2.2.3 The system shall notify users when other users share notes with them.
* 1.2.2.6 The system shall notify users when other users invite them to an organized in-person meeting.
* 1.2.5.1. The chat features should be unique to the lecture. There is no common channel for the entire community.
* 1.2.5.2. Learners can be in contact with lecturers through chat.
* 1.2.5.3. The chat should include the feature of creating events.
* 1.2.5.4. The chat should include a timestamp for each message and the senders' name should be at top of the each message.
* 1.2.5.5. The sender's name should direct the user to his/her profile page.
* 1.2.5.6. The chat messages should be stored.

I will try to cover these requirements by the use cases.

**Comment by egemenatikk** [comment]
I have finished chat related part of the [use case diagram](https://lucid.app/lucidchart/975d6056-2874-4fb9-8642-fc1a3a88aab4/edit?invitationId=inv_dfe17a81-20af-4a01-9d13-dbcc4f7ade2c&page=0_0#). This issue will be reviewed and if necessary edited/updated further.

**Comment by xltvy** [comment]
Thank you for your comprehensive work! I have reviewed the chat section of the use-case diagram. Here are some points I would like to mention:
* I think we may add a *text* option with includes tag to the message. Since, we have provided timestamp, and sender's name as what message includes, I think it also includes the caption as a text.
* Notification's includes are a bit wordy. I think we may shorten them up by deleting notifications at the start.
* If we think that sharing notes are directly something else than sending a message, then this structure is just fine. But if we send a note inside a message, I think we should draw its arrow from **message** not **chat**

**Comment by egemenatikk** [comment]
Thank you for your review. I have changed the parts you mentioned as you suggested. 

For the third list item in your review, I could not come up with an answer. I will post that question to our communication channel and make the necessary changes if majority of our group suggests so.

**Comment by egemenatikk** [comment]
We have discussed the topic above, and decided that notifications are handled by system, not the users themselves. So, we decided to cut that part out of the chat related use case diagram. Here is the final state screenshot of chat related use case diagram that I have prepared:

<img width="auto" alt="chat" src="https://user-images.githubusercontent.com/100771925/161426149-459b5a63-c102-4ac1-9938-60838c582814.png">

**Comment by xltvy** [comment]
After a final review of this issue, I think we can happily close it. Thank you for your efforts.
